docs/README.md

```diff
@@ -0,0 +1,208 @@
+
+### Hubs
+A Hub is much like a Cloud Agent, but rather than focusing only on messaging (transport) as defined above for a Cloud Agent, the Hub also stores and shares data (potentially including Verifiable Credentials), on behalf of its owner. All of the data held by the Hub is en/decrypted by the Edge Agent, so it is the data that moves between the Edge and Hub, and not keys.  The Hub storage can (kind of) be thought of as a remote version of a Wallet without the keys, but is intended to hold more than just the Verifiable Credentials of an Edge Agent wallet. The idea is that the user can push lots of, for example, app-related data to the Hub, and a Service would be granted permission by the Owner to directly access the data without having to go to the Edge Agent. For example, a Hub-centric music service would store the owner’s config information and playlists on the Hub, and the Service would fetch the data from the Hub on use instead of storing it on it’s own servers.
+
+## Messaging Protocol
+### Transport Payload - Message Packaging/Unpackaging
+The process of handling messages should be consistent and independent of the transport mechanism of an agent implementation. The payload put onto the transport should be created in the following way:
+* Anon-encrypt the base message structure
+* [Base64](https://tools.ietf.org/html/rfc4648) encode the encrypted message
+
+The point in the process after the transport mechanism has delivered the payload should also be consistent and follow these steps:
+
+* [Base64](https://tools.ietf.org/html/rfc4648) decode the string
+* Anon-decrypt the bytes
+
+In order to anon encrypt/decrypt the payload, the endpoint (transport) mechanism MUST use a well known DID/VerKey associated with that endpoint called an Endpoint DID. No matter what kind of transport is used, the Endpoint DID should be used to package/unpackage the payload through/from the transport by following the steps outlined above.
+
+### Base/Core Message Structure
+To maintain consistency of message handling, the following defines the structure of every message before it is packaged and sent over the transport layer or after it is received through the transport layer and unpackaged:
+```json
+{
+  "id": "identifier/DID/nonce",
+  "type": "URN message type",
+  "message": {}
+}
+```
+* The `id` attribute is required and needs to be either the DID of the sender that is a pairwise identifier in an established connection, a nonce used in establishing a connection, or another similar identifier used in a different custom message exchange.
+* The `type` attribute is required and MUST be a recognized message type.
+* The `message` attribute is required or optional depending upon the value of the type attribute and would contain the contents specified in the definition of the message type.
+ 
+### Message Types
+Message types MUST be in the form of a URN. For example, Sovrin implementations of Indy agents follow the structure of the URN as:
```

Comment:
I may be mistaken but I thought that the exact form of message types had not yet been decided. If it has been, I think having a reference to that decision somewhere would be good to back up the use of an all capital "MUST."

Reply:
The discussion we had in the Friday, June 15, 2018 agent call had consensus to follow this outlined message protocol.